recipe/meta.yaml

```diff
@@ -0,0 +1,143 @@
+{% set name = "torchaudio" %}
+{% set version = "2.1.2" %}
+{% set run_unittests = False %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pytorch/audio/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 82c2745a73172b495327ec36c6970ad5ad9d5d2ac44feeaea2617152f9393bf7
+
+build:
+  number: 0
+  skip: True  # [py>311]
+  script:
+    - export USE_FFMPEG=ON                         # [not win]
+    - set USE_FFMPEG=ON                            # [win]
+    - export BUILD_SOX=OFF                         # [not win]
+    - set BUILD_SOX=OFF                            # [win]
+    - export USE_OPENMP=ON                         # [not win]
+    - set USE_OPENMP=ON                            # [win]
+    - export BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON  # [not win]
+    - set BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON     # [win]
+    - export FFMPEG_ROOT=${PREFIX}                 # [not win]
+    - set FFMPEG_ROOT=%LIBRARY_PREFIX%             # [win]
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+    - make  # [not win]
+    - patchelf  # [linux]
+    - protobuf {{ libprotobuf }}
+    - llvm-openmp  # [osx]
+    - intel-openmp  # [not osx and blas_impl == 'mkl']
+  host:
+    - python
+    - setuptools
+    - wheel
+    - pip
+    - pytorch 2.1.0
+    - pybind11
+    - ffmpeg >=6,<7
```

Comment:
ffmpeg should be pinned exactly in host and then pin_compatible in run (although there's a run_export in ffmpeg so that's not needed)

Reply:
what about the version restriction though? Maybe I'm not too familiar with the inner workings of `pin_compatible`. But if `ffmpeg 6.1.1` is pinned in host, how does `pin_compatible` know that because of API breakage of major releases?

Reply:
https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#pinning-at-the-recipe-level

The docs can be referred to for this kind of understanding.